src/osiris_log.erl

```diff
@@ -1903,7 +1903,7 @@ update_retention(Retention,
 evaluate_retention(Dir, Specs) when is_list(Dir) ->
     % convert to binary for faster operations later
     % mostly in segment_from_index_file/1
-    evaluate_retention(list_to_binary(Dir), Specs);
+    evaluate_retention(filename:join(Dir, <<"">>), Specs);
```

Comment:
Should we explain why we use this exotic way of converting a string into a binary?